mixer/adapter/dogstatsd/dogstatsd.go

```diff
@@ -0,0 +1,206 @@
+	for _, m := range ac.Metrics {
+		if m.Tags == nil {
+			m.Tags = map[string]string{}
+		}
+	}
+	return &handler{ac.SampleRate, client, ac.Metrics}, nil
+}
+
+// adapter.HandlerBuilder#SetAdapterConfig
+func (b *builder) SetAdapterConfig(cfg adapter.Config) { b.adapterConfig = cfg.(*config.Params) }
+
+// adapter.HandlerBuilder#Validate
+func (b *builder) Validate() (ce *adapter.ConfigErrors) {
+	ac := b.adapterConfig
+
+	// Validate the buffer is a valid size
+	if ac.BufferLength < 0 {
+		ce = ce.Appendf("bufferLength", "buffer length must be >= 0")
+	}
+
+	// Validate the sample rate is a valid percentage value
+	if ac.SampleRate < 0 || ac.SampleRate > 1 {
+		ce = ce.Appendf("sampleRate", "sampling rate must be between 0 and 1")
+	}
+
+	// Validate the address of the agent is valid
+	if ac.Address == "" {
+		ce = ce.Appendf("address", "Address is empty")
+	}
+	if _, _, err := net.SplitHostPort(ac.Address); err != nil {
+		ce = ce.Appendf("address", "Address is malformed: %v", err)
+	}
+
+	// Validate the adapter handles all metrics it is being sent
+	for mname := range b.metricTypes {
+		if _, found := ac.Metrics[mname]; !found {
+			ce.Appendf("metricName", "%s is a valid metric but is not configured to be handled byt the datadog adapter", mname)
+		}
+	}
+
+	// Validate metrics that will be handled are of the right type and that they exist
+	for mname, metric := range ac.Metrics {
+		m, found := b.metricTypes[mname]
+		if !found {
+			ce = ce.Appendf("metricType", "%s is not a valid metric that can be emitted", mname)
+		} else {
+			val := m.Value
+			switch metric.Type {
+			case config.COUNTER, config.GAUGE:
+				if val != descriptor.INT64 {
+					ce = ce.Appendf("metricType", "Counters and Gauges must be an Int64 but metric %s is a %s", mname, val.String())
+				}
+			case config.DISTRIBUTION:
+				if val != descriptor.DURATION && val != descriptor.INT64 {
+					ce = ce.Appendf("metricType", "Histograms must be either a Duration or an Int64 but metric %s is a %s", mname, val.String())
+				}
+			default:
+				ce = ce.Appendf("metricType", "%s is not a supported metric type for this adapter.", val.String())
+			}
+		}
+	}
+	return ce
+}
+
+// metric.HandlerBuilder#SetMetricTypes
+func (b *builder) SetMetricTypes(types map[string]*metric.Type) {
+	b.metricTypes = types
+}
+
+////////////////// Request-time Methods //////////////////////////
+// metric.Handler#HandleMetric
+func (h *handler) HandleMetric(_ context.Context, insts []*metric.Instance) error {
+	var result *multierror.Error
+	for _, inst := range insts {
+		if err := h.record(inst); err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+	return result.ErrorOrNil()
+}
+
+func (h *handler) record(value *metric.Instance) error {
+	mname := value.Name
+
+	// Shouldn't fail since validation checks metric creation
+	info, _ := h.metrics[mname]
+
+	// Add dimensions to tags
+	for k, v := range value.Dimensions {
+		info.Tags[k] = fmt.Sprintf("%v", v)
```

Comment:
@awwithro you should now be able to replace this code with the new code from pkg/adapter.Stringify

Reply:
will do. For purposes of rebasing/pushing is the PR "good enough" to force push and lose comment history?